44.md

```diff
@@ -0,0 +1,167 @@
+NIP-44
+======
+
+Encrypted Payloads (Versioned)
+------------------------------
+
+`optional` `author:paulmillr` `author:staab`
+
+The NIP introduces a new data format for keypair-based encryption. This NIP is versioned to allow multiple algorithm choices to exist simultaneously.
+
+The encrypted payload is encoded as base64. The first byte of the payload represents the version. The rest depends on the version.
+
+Currently defined encryption algorithms:
+
+- `0x00` - Reserved
+- `0x01` - XChaCha with same key `sha256(ecdh)` per conversation
```

Comment:
XChaCha20 (add the '20' to the end)

Also what is this "per conversation"?

Reply:
It's common to call it just ChaCha, instead of Salsa20 - which is usually suffixed with 20. So it's cool I think.

Reply:
But there are 8 and 12 round versions. Crypto people may know that 20 is implied, but nostr devs may stumble upon the other ones and not know which one to use. See this lib for example: https://docs.rs/chacha20/latest/chacha20/